src/mmap_unix.rs

```diff
@@ -74,6 +74,25 @@ impl error::Error for Error {}
 
 pub type Result<T> = result::Result<T, Error>;
 
+pub(crate) fn mmap_unix(
+    size: usize,
+    prot: i32,
+    flags: i32,
+    fd: i32,
+    f_off: u64,
+) -> Result<*mut u8> {
```

Comment:
I wonder: If we now add more dynamic mmap behaviour, would it maybe make sense to wrap the result pointer in some kind of guard object that can do automatic unmapping if the object is dropped? My feeling is that this may also provide some options to simplify the `unsafe`-use that will be required to use this return pointer.